File: Sparse Arrays/sparse-arrays.py
# Complete the matchingStrings function below.
def matchingStrings(strings, queries):
    """
    :type strings: List[str]
    :type queries: List[str]
    :rtype: List[int]
    """
    hash = {}
    for s in strings:
        hash[s] = hash[s] + 1 if s in hash else 1

    res = []
    for q in queries:
        res.append(hash.get(q) if q in hash else 0)

    return res


print(matchingStrings(['aba', 'baba', 'aba', 'xzxb'], ['aba', 'xzxb', 'ab']))  # [2, 1, 0]

File: Sparse Arrays/sparse-arrays.php
<?php
/**
 * Created by PhpStorm.
 * User: kovalevcon
 * Date: 27.09.18
 * Time: 18:30
 */

/**
 * Sparse Arrays function
 *
 * @param array $strings
 * @param array $queries
 * @return array
 */
function matchingStrings(array $strings, array $queries): array
{
    $hash = [];
    foreach ($strings as $s) {
        isset($hash[$s]) ? $hash[$s]++ : $hash[$s] = 1;
    }

    $res = [];
    foreach ($queries as $q) {
        array_push($res, $hash[$q] ?? 0);
    }

    return $res;
}


echo "<pre>matchingStrings(['aba', 'baba', 'aba', 'xzxb'], ['aba', 'xzxb', 'ab'])\r\n";
var_dump(matchingStrings(['aba', 'baba', 'aba', 'xzxb'], ['aba', 'xzxb', 'ab'])); // [2, 1, 0]
echo "</pre>";
